js/runner_test.go

```diff
@@ -151,6 +152,53 @@ func TestRunnerOptions(t *testing.T) {
 	}
 }
 
+func TestRunnerRPSLimit(t *testing.T) {
+	t.Parallel()
+
+	var nilLimiter *rate.Limiter
+
+	variants := []struct {
+		name    string
+		options lib.Options
+		limiter *rate.Limiter
+	}{
+		{
+			name:    "RPS not defined",
+			options: lib.Options{},
+			limiter: nilLimiter,
+		},
+		{
+			name:    "RPS set to non-zero int",
+			options: lib.Options{RPS: null.IntFrom(9)},
+			limiter: rate.NewLimiter(rate.Limit(9), 1),
+		},
+		{
+			name:    "RPS set to zero",
+			options: lib.Options{RPS: null.IntFrom(0)},
+			limiter: nilLimiter,
+		},
+		{
+			name:    "RPS set to below zero value",
+			options: lib.Options{RPS: null.IntFrom(-1)},
+			limiter: nilLimiter,
+		},
+	}
+
+	for _, variant := range variants {
+		variant := variant
+
+		t.Run(variant.name, func(t *testing.T) {
+			t.Parallel()
+
+			r, err := getSimpleRunner(t, "/script.js", `exports.default = function() {};`)
+			require.NoError(t, err)
+			err = r.SetOptions(variant.options)
+			require.NoError(t, err)
+			assert.Equal(t, variant.limiter, r.RPSLimit)
```

Comment:
hmm I am kind of surprised this works correctly for the `non-zero int` case :confused: I am not sure if testify is actually comparing the struct contents behind the pointers or what :thinking: 

in any case, it might be good enough to just compare if the limiter is `nil` or not `nil`

Reply:
From the [docs](https://pkg.go.dev/github.com/stretchr/testify/require#Equal)

> Pointer variable equality is determined based on the equality of the referenced values (as opposed to the memory addresses). Function equality cannot be determined and will always fail. 

So this seems like it should work and apperantly it does.